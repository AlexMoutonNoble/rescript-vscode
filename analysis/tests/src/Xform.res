type kind = First | Second | Third
type r = {name: string, age: int}

let ret = _ => assert false
let kind = assert false

if kind == First {
  // ^xfm
  ret("First")
} else {
  ret("Not First")
}

#kind("First", {name: "abc", age: 3}) != kind ? ret("Not First") : ret("First")
//             ^xfm

let name = "hello"
//   ^xfm

let annotated: int = 34
//   ^xfm

module T = {
  type r = {a: int, x: string}
}

let foo = x =>
  switch x {
  | None => 33
  | Some(q) => q.T.a + 1
  //     ^xfm
  }
