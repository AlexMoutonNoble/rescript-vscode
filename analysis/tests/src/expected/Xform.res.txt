Xform tests/src/Xform.res 6:5
Hit: Replace with switch
{"start": {"line": 6, "character": 0}, "end": {"line": 11, "character": 1}}
newText:
switch kind {
| First =>
  // ^xfm
  ret("First")
| _ => ret("Not First")
}

Xform tests/src/Xform.res 13:15
Hit: Replace with switch
{"start": {"line": 13, "character": 0}, "end": {"line": 13, "character": 79}}
newText:
switch kind {
| #kind("First", {name: "abc", age: 3}) => ret("First")
| _ => ret("Not First")
}

Xform tests/src/Xform.res 16:5
Hit: Add type annotation
{"start": {"line": 16, "character": 8}, "end": {"line": 16, "character": 8}}
newText:
: string

Xform tests/src/Xform.res 19:5

Xform tests/src/Xform.res 26:10
Hit: Add type annotation
{"start": {"line": 26, "character": 10}, "end": {"line": 26, "character": 11}}
newText:
(x: option<T.r>)

Xform tests/src/Xform.res 30:9

Xform tests/src/Xform.res 34:21
Hit: Add type annotation
{"start": {"line": 34, "character": 24}, "end": {"line": 34, "character": 24}}
newText:
: int

Xform tests/src/Xform.res 38:5

Xform tests/src/Xform.res 41:9
Hit: Add type annotation
{"start": {"line": 41, "character": 11}, "end": {"line": 41, "character": 11}}
newText:
: int

